### PR TITLE
Network shifter: connect transformers for ES & PT scalable generators

### PR DESCRIPTION
**Pull request by pet-mit**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
When a ES or PT scalable is connected to grid through a disconnected transformer, it cannot be scaled.

**What is the new behavior (if this is a feature change)?**
Before scaling, transformers are connected. After scaling, transformers of generators that were not scaled are re-disconnected.

### CODE REVIEW

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/ScalableGeneratorConnector.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ */
+public class ScalableGeneratorConnector {
+    private Map<String, GeneratorState> changedGeneratorsInitialState = new HashMap<>();
+    private final ZonalData<Scalable> zonalScalable;
+
+    ScalableGeneratorConnector(ZonalData<Scalable> zonalScalable) {
+        this.zonalScalable = zonalScalable;
+    }
+
+    void prepareForScaling(Network network) throws ShiftingException {
```
Comment (phiedw) [inline gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/ScalableGeneratorConnector.java:34]: do we want to add the countries as parameters of this method for more genericity? (and so it's a bit clearer what's being connected when we call this method)

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/ScalableGeneratorConnector.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                && !getBus(gen.getTerminal()).isInMainConnectedComponent()).collect(Collectors.toSet());
+        for (Generator gen : generators) {
+            connectGeneratorTwoWindingsTransformer(gen, network);
+        }
+    }
+
+    /**
+     * Checks if a generator has exactly one transformer connecting it to the main network and connects it.
+     * (If no transformer exist, nothing is modified)
+     * (If more than one transformer exist, TODO)
```
Comment (phiedw) [inline gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/ScalableGeneratorConnector.java:62]: Do we abandon this TODO for now or are we waiting for Mario's answer?

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/SweNetworkShifter.java [hunk 3]
```diff
@@ -149,13 +152,24 @@ public void shiftNetwork(double stepValue, Network network) throws GlskLimitatio
 
             // Step 5: Reset current variant with initial state
             network.getVariantManager().setWorkingVariant(initialVariantId);
+            network.getVariantManager().removeVariant(processedVariantId);
             network.getVariantManager().removeVariant(workingVariantCopyId);
         } finally {
+            // revert connections of TWT on generators that were not used by the scaling
+            scalableGeneratorConnector.revertUnnecessaryChanges(network);
```
Comment (phiedw) [inline gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/SweNetworkShifter.java:159]: Does setting the variant to the initialVariant not already reset terminal connections and targetP?

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/ScalableGeneratorConnector.java (hunk 2 above)
Comment (pet-mit) [reply gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/ScalableGeneratorConnector.java:62]: awaiting Mario's answer

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/SweNetworkShifter.java (hunk 3 above)
Comment (pet-mit) [reply gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/SweNetworkShifter.java:159]: we want to revert only the ones that were not connected by scaling and set to a specific targetP

**Pull request merged**